src/platform/with_wgl/utils.rs

```diff
@@ -0,0 +1,703 @@
+use std::mem;
+use super::wgl_attributes::*;
+
+use std::ffi::{CStr, CString, OsStr};
+use std::os::raw::{c_void, c_int};
+use std::os::windows::ffi::OsStrExt;
+use std::io;
+use std::ptr;
+
+use winapi;
+use kernel32;
+use user32;
+use gdi32;
+use super::wgl;
+use super::wgl_ext;
+
+pub unsafe fn create_offscreen(shared_with: winapi::HGLRC,
+                               settings: &WGLAttributes)
+                               -> Result<(winapi::HGLRC, winapi::HDC), String> {
+
+    let window: winapi::HWND = try!(create_hidden_window());
+    let hdc = user32::GetDC(window);
+    if hdc.is_null() {
+        return Err("GetDC function failed".to_owned());
+    }
+
+    let extra = try!(load_extra_functions(window));
+
+    let extensions = if extra.GetExtensionsStringARB.is_loaded() {
+        let data = extra.GetExtensionsStringARB(hdc as *const _);
+        let data = CStr::from_ptr(data).to_bytes().to_vec();
+        String::from_utf8(data).unwrap()
+
+    } else if extra.GetExtensionsStringEXT.is_loaded() {
+        let data = extra.GetExtensionsStringEXT();
+        let data = CStr::from_ptr(data).to_bytes().to_vec();
+        String::from_utf8(data).unwrap()
+
+    } else {
+        format!("")
+    };
+
+
+    let (id, _) = if extensions.split(' ').find(|&i| i == "WGL_ARB_pixel_format").is_some() {
+        try!(choose_arb_pixel_format(&extra, &extensions, hdc, &settings.pixel_format)
+            .map_err(|_| "Pixel format not available".to_owned()))
+    } else {
+        try!(choose_native_pixel_format(hdc, &settings.pixel_format)
+            .map_err(|_| "Pixel format not available".to_owned()))
+    };
+
+    try!(set_pixel_format(hdc, id));
+
+    create_full_context(settings, &extra, &extensions, hdc, shared_with)
+
+}
+
+// creates a basic context
+unsafe fn create_basic_context(hdc: winapi::HDC,
+                               share: winapi::HGLRC)
+                               -> Result<(winapi::HGLRC, winapi::HDC), String> {
+    let ctx = wgl::CreateContext(hdc as *const c_void);
+    if ctx.is_null() {
+        return Err(format!("wglCreateContext failed: {}",
+                           format!("{}", io::Error::last_os_error())));
+    }
+
+    if !share.is_null() {
+        if wgl::ShareLists(share as *const c_void, ctx) == 0 {
+            return Err(format!("wglShareLists failed: {}",
+                               format!("{}", io::Error::last_os_error())));
+        }
+    };
+
+    return Ok((ctx as winapi::HGLRC, hdc));
+}
+
+// creates a full context: attempts to use optional ext WGL functions
+unsafe fn create_full_context(settings: &WGLAttributes,
```

Comment:
This seems sensible enough, though, I'll need to study a bit more since I'm not confident enough with the windows API. Thanks for pointing to glutin.

Also, can you add proper attribution to the `glutin` code?